specification/synapse/resource-manager/Microsoft.Synapse/preview/2021-04-01-preview/kustoPool.json

```diff
@@ -0,0 +1,1534 @@
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/CloudError"
+            }
+          }
+        }
+      },
+      "put": {
+        "tags": [
+          "kustoPools"
+        ],
+        "operationId": "kustoPools_CreateOrUpdate",
+        "x-ms-examples": {
+          "kustoPoolsCreateOrUpdate": {
+            "$ref": "./examples/KustoPoolsCreateOrUpdate.json"
+          }
+        },
+        "x-ms-long-running-operation": true,
+        "description": "Create or update a Kusto pool.",
+        "parameters": [
+          {
+            "$ref": "./workspace.json#/parameters/workspaceNameParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/kustoPoolNameParameter"
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/KustoPool"
+            },
+            "description": "The Kusto pool parameters supplied to the CreateOrUpdate operation."
+          },
+          {
+            "name": "If-Match",
+            "in": "header",
+            "required": false,
+            "type": "string",
+            "x-ms-client-name": "IfMatch",
+            "description": "The ETag of the Kusto Pool. Omit this value to always overwrite the current Kusto Pool. Specify the last-seen ETag value to prevent accidentally overwriting concurrent changes."
+          },
+          {
+            "name": "If-None-Match",
+            "in": "header",
+            "required": false,
+            "type": "string",
+            "x-ms-client-name": "IfNoneMatch",
+            "description": "Set to '*' to allow a new Kusto Pool to be created, but to prevent updating an existing Kusto Pool. Other values will result in a 412 Pre-condition Failed response."
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Successfully updated the Kusto Pool.",
+            "schema": {
+              "$ref": "#/definitions/KustoPool"
+            }
+          },
+          "201": {
+            "description": "Successfully created the Kusto Pool.",
+            "schema": {
+              "$ref": "#/definitions/KustoPool"
+            }
+          },
+          "202": {
+            "description": "Successfully updated the Kusto Pool.",
+            "schema": {
+              "$ref": "#/definitions/KustoPool"
+            }
+          },
```

Comment:
For the long running PUT operations, 201+ provisioningState response model is recommended. Please refer to the the RPC document here: [https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/Addendum.md#creating-or-updating-resources-asynchronously](https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/Addendum.md#creating-or-updating-resources-asynchronously)

Reply:
Modified all PUT operations accordingly.

Reply:
It looks like the status code descriptions are updated. However, it seems like there are two main problems here 1) 202 response has a body (KustoPool) which is not supported 2) The response model is in between the supported 201+provisioningState and 202+location headers. I see that kusto.json (which you refered before) has 201+provisioningState - did you mean to support the same pattern? [https://github.com/Azure/azure-rest-api-specs/blob/1b0ed8edd58bb7c9ade9a27430759527bd4eec8e/specification/azure-kusto/resource-manager/Microsoft.Kusto/stable/2020-09-18/kusto.json#L94](https://github.com/Azure/azure-rest-api-specs/blob/1b0ed8edd58bb7c9ade9a27430759527bd4eec8e/specification/azure-kusto/resource-manager/Microsoft.Kusto/stable/2020-09-18/kusto.json#L94) 

---
In reply to: [612535339](https://github.com/Azure/azure-rest-api-specs/pull/13874#discussion_r612535339) [](ancestors = 612535339)

Reply:
On point - removed.